docs/reference/provisioning-methods.rst

```diff
@@ -0,0 +1,74 @@
+Provisioning Methods
```

Comment:
I think only this file actually really belongs in the testflinger docs as a reference, the others are a straightforward fit to Library.

This indeed should also be a _reference_ so that there's an example configuration included (and a hello-world type payload at the top of the file indicating where the choice of the provisioning method goes?)